### PR TITLE
Add IntelliJ instructions

### PR DESCRIPTION
**Pull request by oliver-amzn**
As per our discussion yesterday, adding a few instructions to the doc so they're not getting lost before the next person onboards.

---
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)

### CODE REVIEW

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 1 unresolved discussion (waiting on @corbantek, @oliver-amzn, @stevenwarejones, and @zachcwc)

___
*[docs/building.md, line 34 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101#-Mbm1O_U19vU9cWdoOeu:-Mbm1O_U19vU9cWdoOev:b-uatyl) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/622739a83b662ea1b31405dba8927beaf931a563/docs/building.md#L34)):*
> ```Markdown
> `tools/bazel-container build "//..."`
> 
> As of early June 2021, these steps work to get code completion working for Intellij on MacOS:
> ```

I think we want to separate setting up IntelliJ elsewhere as an optional helpful thing, and keep this as a discussion of building on MacOS. For example, describing that you can build many of the targets natively with Bazel on MacOS, but that there are some that require Linux at the moment.

Wherever we document IntelliJ project setup, we can mention that target compatibility should be sufficient to develop on MacOS, but not all tests will build and refer back to this section.

**Review by oliver-amzn — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 1 unresolved discussion (waiting on @corbantek, @SanjayVas, @stevenwarejones, and @zachcwc)

___
*[docs/building.md, line 34 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101#-Mbm1O_U19vU9cWdoOeu:-MbmEqnM-FzqIqZd73XJ:belbqia) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/622739a83b662ea1b31405dba8927beaf931a563/docs/building.md#L34)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I think we want to separate setting up IntelliJ elsewhere as an optional helpful thing, and keep this as a discussion of building on MacOS. For example, describing that you can build many of the targets natively with Bazel on MacOS, but that there are some that require Linux at the moment.

Wherever we document IntelliJ project setup, we can mention that target compatibility should be sufficient to develop on MacOS, but not all tests will build and refer back to this section.
</blockquote></details>

Eh, I disagree. I wouldn't want to fragment the documentation too much considering how little we have right now and IMO this is related closely to this section so that someone that starts working on MacOS would find it here.

This is block is of temporary nature until we have a proper MacOS setup but this will need to wait a little due to bandwidth constraints :-/

File: docs/building.md [hunk 1]
```diff
@@ -31,6 +31,14 @@ bazel test //src/test/...
...
 `tools/bazel-container build "//..."`
 
+As of early June 2021, these steps work to get code completion working for Intellij on MacOS:
+- use IntelliJ 2021.1.2 and the beta build of the Bazel Plugin
+    - Add `https://plugins.jetbrains.com/plugins/list?channel=beta` to your list of plugin repos ([source](https://github.com/bazelbuild/intellij/issues/2406))
+- Start a new project from scratch in IntelliJ by selecting "Import Bazel Project" and open the base dir of the cross-media-measurement repo
+    - Uncomment "Kotlin" when creating the .bazelproject file
+- Once the new projected is created, the Bazel plugin will kick off a sync and, after a while, IntelliJ should start detecting all dependencies and index them. Ignore the Bazel output errors, they happen because e.g. the Spanner Emulator can only be build on Linux.
+- You can run most Kotlin tests (with some exceptions, e.g. TransportSecurityTest.kt doesn't work because of openssl incompatibilities) by directly invoking a particular test: `bazel test "//src/test/kotlin/org/wfanet/measurement/common/identity/..."` or similar.
+
```
Comment (corbantek) [inline docs/building.md:41]: Maybe add a quick note stating that using the bazel-container build above is in somewhat contrast at the moment with IntelliJ setup on Mac (and will most likely temporary break IntelliJ until you "resync" again)

Comment (oliver-amzn) [reply docs/building.md:41]: Yeah, that's a good idea.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 2 unresolved discussions (waiting on @corbantek, @oliver-amzn, @stevenwarejones, and @zachcwc)

___
*[docs/building.md, line 34 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101#-Mbm1O_U19vU9cWdoOeu:-MbmpbRC8FfbFXF5hilD:bb4wbot) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/622739a83b662ea1b31405dba8927beaf931a563/docs/building.md#L34)):*
<details><summary><i>Previously, oliver-amzn (Oliver) wrote…</i></summary><blockquote>

Eh, I disagree. I wouldn't want to fragment the documentation too much considering how little we have right now and IMO this is related closely to this section so that someone that starts working on MacOS would find it here.

This is block is of temporary nature until we have a proper MacOS setup but this will need to wait a little due to bandwidth constraints :-/
</blockquote></details>

I just want to separate the IntelliJ project setup instructions since they apply to both Linux and MacOS. Note that "Wherever we document IntelliJ project setup" could be another section here in this file.

**Review by oliver-amzn — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @corbantek, @SanjayVas, @stevenwarejones, and @zachcwc)

___
*[docs/building.md, line 34 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101#-Mbm1O_U19vU9cWdoOeu:-Mbqi8tzCluxJHuZP5pu:b-7qfuwa) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/622739a83b662ea1b31405dba8927beaf931a563/docs/building.md#L34)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I just want to separate the IntelliJ project setup instructions since they apply to both Linux and MacOS. Note that "Wherever we document IntelliJ project setup" could be another section here in this file.
</blockquote></details>

That makes sense. Broke it out into a separate section.

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 2 unresolved discussions (waiting on @corbantek, @SanjayVas, @stevenwarejones, and @zachcwc)

**Review by SanjayVas — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 1 unresolved discussion (waiting on @corbantek, @stevenwarejones, and @zachcwc)

**Review by zachcwc — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 1 unresolved discussion (waiting on @corbantek and @stevenwarejones)

**Review by oliver-amzn — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: all files reviewed, 1 unresolved discussion (waiting on @corbantek and @stevenwarejones)

___
*[docs/building.md, line 41 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101#-Mblza-W6eZFI-8z-u91-r1-41:-Mbvs3RQDG3KGvjVpV8C:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/622739a83b662ea1b31405dba8927beaf931a563/docs/building.md#L41)):*
<details><summary><i>Previously, oliver-amzn (Oliver) wrote…</i></summary><blockquote>

Yeah, that's a good idea.
</blockquote></details>

Done.

**Review by corbantek — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/101)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @stevenwarejones)

**Pull request merged**